asv_bench/benchmarks/ctors.py

```diff
@@ -3,17 +3,45 @@
 from pandas import Series, Index, DatetimeIndex, Timestamp, MultiIndex
 
 
+def no_change(arr):
+    return arr
+
+
+def list_of_str(arr):
+    return list(arr.astype(str))
+
+
+def arr_dict(arr):
+    return dict(zip(range(len(arr)), arr))
+
+
+def list_of_tuples(arr):
+    return [(i, -i) for i in arr]
+
+
+def list_of_lists(arr):
+    return [[i, -i] for i in arr]
+
+
+def list_of_tuples_with_none(arr):
```

Comment:
Naming nit but this is a generator expression not a list, no?

Reply:
Done - generators don't support `+`, so didn't add them for this particular case